File: app/src/main/java/ru/melod1n/schedule/LoginActivity.java
package ru.melod1n.schedule;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;

import com.google.android.material.snackbar.Snackbar;

import butterknife.BindView;
import butterknife.ButterKnife;
import ru.melod1n.schedule.common.ThemeEngine;
import ru.melod1n.schedule.current.BaseActivity;
import ru.melod1n.schedule.widget.Button;
import ru.melod1n.schedule.widget.TextArea;
import ru.melod1n.schedule.widget.Toolbar;

public class LoginActivity extends BaseActivity {

    private static final int REQUEST_REGISTRATION = 1;

    @BindView(R.id.button_registration)
    Button buttonRegistration;

    @BindView(R.id.button_login)
    Button buttonLogin;

    @BindView(R.id.input_login)
    TextArea inputLogin;

    @BindView(R.id.input_password)
    TextArea inputPassword;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        ButterKnife.bind(this);

        toolbar.setTitle(R.string.login);
        toolbar.setNavigationIcon(R.drawable.ic_arrow_backward);
        toolbar.setNavigationOnClickListener(v -> onBackPressed());

        buttonRegistration.setOnClickListener(v -> openRegistrationScreen());
        buttonLogin.setOnClickListener(v -> tryLogin());
    }

    private void tryLogin() {
        String login = inputLogin.getText().toString();
        String password = inputPassword.getText().toString();

        try {
            int i = 1;
            i /= Integer.parseInt(password);

            i = 1 / Integer.parseInt(login);

            runOnUiThread(() -> {
                setResult(RESULT_OK); //проверка пройдена, успех
                finish();
            });
        } catch (Exception e) {
            runOnUiThread(() -> displayError("Hueta"));
        }

    }

    private void displayError(String description) {
        Snackbar.make((View) inputLogin.getParent(), String.format("%s: %s", getString(R.string.error), description), Snackbar.LENGTH_LONG)
                .setAction(R.string.registration, v -> openRegistrationScreen())
                .setActionTextColor(ThemeEngine.getCurrentTheme().getColorAccent())
                .show();
    }

    private void openRegistrationScreen() {
        startActivityForResult(new Intent(this, RegistrationActivity.class), REQUEST_REGISTRATION);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);

        if (requestCode == REQUEST_REGISTRATION && resultCode == RESULT_OK) { //успешно зарегались
            setResult(RESULT_OK); //успешно авторизовались
            finish();
        }
    }
}

File: app/src/main/java/ru/melod1n/schedule/activity/LoginActivity.kt
package ru.melod1n.schedule.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.toolbar.*
import ru.melod1n.schedule.R
import ru.melod1n.schedule.base.BaseActivity

class LoginActivity : BaseActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        toolbar.setTitle(R.string.login)
        toolbar.setNavigationIcon(R.drawable.ic_arrow_backward)
        toolbar.setNavigationOnClickListener { onBackPressed() }

        buttonRegistration.setOnClickListener { openRegistrationScreen() }
        buttonLogin.setOnClickListener { tryLogin() }
    }

    private fun tryLogin() {
        val login = inputLogin!!.text.toString()
        val password = inputPassword!!.text.toString()

        try {
            var i = 1
            i /= password.toInt()
            i = 1 / login.toInt()
            runOnUiThread {
                setResult(RESULT_OK) //проверка пройдена, успех
                finish()
            }
        } catch (e: Exception) {
            runOnUiThread { displayError("Hueta") }
        }
    }

    private fun displayError(description: String) {
        Snackbar.make((inputLogin.parent as View), String.format("%s: %s", getString(R.string.error), description), Snackbar.LENGTH_LONG)
                .setAction(R.string.registration) { openRegistrationScreen() }
//                .setActionTextColor(currentTheme!!.colorAccent)
                .show()
    }

    private fun openRegistrationScreen() {
        startActivityForResult(Intent(this, RegisterActivity::class.java), REQUEST_REGISTRATION)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == REQUEST_REGISTRATION && resultCode == RESULT_OK) { //успешно зарегались
            setResult(RESULT_OK) //успешно авторизовались
            finish()
        }
    }

    companion object {
        private const val REQUEST_REGISTRATION = 1
    }
}